### DISCUSSION TITLE
BnN Decrypting

### DISCUSSION THREAD

**Discussion by Prysmarine** [noDRM/DeDRM_tools#629] [Q&A] [answered]
I used DeDRM earlier this year to move around my nook library without issue. I'm doing a big redo of my entire library and I'm basically starting over from scratch, but for some reason the nook books are no longer working when importing into Calibre. I haven't updated Calibre or DeDRM, they're the same versions that I was using last time: Calibre 7.9.0 and DeDRM 10.0.3, and it's the same files I used last time as well. The files have been sitting on a backup hard drive with the Calibre installation and the key that was used to decrypt but when they get added into Calibre they only have the cover page and nothing else.

**Comment by ElleKayEm** [comment]
Don't really understand what is going on.  You have decrypted files but they aren't working in the calibre viewer?

**Reply by Prysmarine** [reply]
I had decrypted files. I am decrypting them again from the same files and password used originally. They are decrypting but only coming out with the cover image and no pages.

**Reply by ElleKayEm** [reply]
I know the calibre viewer will only show the cover with BN comics.  Assume it is because they're set up in some way the calibre viewer doesn't handle.  I've unzipped the epubs and made cbz from the page images.

**Reply by Prysmarine** [reply, marked as answer]
I did a little more messing around with it this morning, apparently something with calibre's conversion from epub to zip is causing the deletion of all the pages. The epub file is decrypted just fine and contains all the pages if I pull it out and manually change it around to look (I apparently never did that yesterday) but I don't know why converting it to a zip with calibre would delete all the pages.

**Reply by ElleKayEm** [reply]
Never used calibre to convert an epub to a zip so I have no idea what that actually does.  An epub is a zip file.  I unzipped the epub, pulled out the page images, and zipped them up to make a cbz.  Then I used a comic book viewer app to read them.

**Reply by Prysmarine** [reply]
It looks like it's changing the file format somewhat, but I'm not entirely sure what it's doing either. I was doing it mostly so that I don't have to manually do it myself for every book I own. If it was just as easy as unzipping the epub and ripping the pages folder I'd be delighted to just do that. Sadly for a lot of manga though there is a blank spacer page in the beginning that gets left out of the images folder but is inserted as part of the xhtml files. If you don't have that blank page it throws off the formatting of the rest of the book making reading it in a two page format very off putting. 

The original problem is handled though so I'll just have to make a small program to handle dealing with BN books and do some small formatting.
